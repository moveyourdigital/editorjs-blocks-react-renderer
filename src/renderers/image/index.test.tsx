import React from 'react';
import { create } from 'react-test-renderer';
import Image, { ImageBlockData } from '.';

describe('<Image />', () => {
  describe('when receives a Image block with actions', () => {
    const data: ImageBlockData = {
      file: {
        url: 'https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg',
        name: '681-2000x1300-2',
      },
      caption: 'Deep in the <b>universe</b>',
      withBorder: true,
      stretched: true,
      withBackground: true,
    };

    it('renders a <figure> block with <img> and <figcaption>', () => {
      expect(create(<Image data={data} className="special-image" />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a Image block without actions', () => {
    const data: ImageBlockData = {
      file: {
        url: 'https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg',
        name: '681-2000x1300-2',
      },
      caption: 'Deep in the universe',
      withBorder: false,
      stretched: false,
      withBackground: false,
    };

    it('renders a <figure> block with <img> and <figcaption>', () => {
      expect(create(<Image data={data} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a SimpleImage block', () => {
    const data: ImageBlockData = {
      url: 'https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg',
      caption: 'Deep in the universe',
      withBorder: false,
      stretched: false,
      withBackground: false,
    };

    it('renders a <figure> block with <img> and <figcaption>', () => {
      expect(create(<Image data={data} />).toJSON()).toMatchSnapshot();
    });
  });



  describe('when receives a Image block', () => {
    const data: ImageBlockData = {
      file: {
        url: '/uploads/2020/08/681-2000x1300-2.jpg',
        name: '681-2000x1300-2',
      },
      caption: 'Deep in the universe',
      withBorder: false,
      stretched: false,
      withBackground: false,
    };

    it('renders with file base href', () => {
      expect(create(<Image data={data} fileBaseHref={'https://google.com'} />).toJSON()).toMatchSnapshot();
    });

    it('renders without file base href', () => {
      expect(create(<Image data={data} />).toJSON()).toMatchSnapshot();
    });
  });
});
