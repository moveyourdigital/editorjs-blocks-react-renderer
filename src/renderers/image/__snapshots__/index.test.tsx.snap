// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> when receives a Image block renders with file base href 1`] = `
<figure>
  <img
    alt="Deep in the universe"
    src="https://google.com/uploads/2020/08/681-2000x1300-2.jpg"
  />
  <figcaption>
    Deep in the universe
  </figcaption>
</figure>
`;

exports[`<Image /> when receives a Image block renders without file base href 1`] = `
<figure>
  <img
    alt="Deep in the universe"
    src="/uploads/2020/08/681-2000x1300-2.jpg"
  />
  <figcaption>
    Deep in the universe
  </figcaption>
</figure>
`;

exports[`<Image /> when receives a Image block with actions renders a <figure> block with <img> and <figcaption> 1`] = `
<figure
  className="special-image image-block--stretched image-block--with-border image-block--with-background"
>
  <img
    alt="Deep in the <b>universe</b>"
    src="https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg"
  />
  <figcaption>
    Deep in the 
    <b>
      universe
    </b>
  </figcaption>
</figure>
`;

exports[`<Image /> when receives a Image block without actions renders a <figure> block with <img> and <figcaption> 1`] = `
<figure>
  <img
    alt="Deep in the universe"
    src="https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg"
  />
  <figcaption>
    Deep in the universe
  </figcaption>
</figure>
`;

exports[`<Image /> when receives a Image block without name renders a <figure> block with <img> and <figcaption> 1`] = `
<figure>
  <img
    alt="Deep in the universe"
    src="https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg"
  />
  <figcaption>
    Deep in the universe
  </figcaption>
</figure>
`;

exports[`<Image /> when receives a SimpleImage block renders a <figure> block with <img> and <figcaption> 1`] = `
<figure>
  <img
    alt="Deep in the universe"
    src="https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg"
  />
  <figcaption>
    Deep in the universe
  </figcaption>
</figure>
`;
